spec/features/hq/create_sponsor_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.feature 'User enters new sponsor data', :type => :feature do
+
+  background do
+    i_sign_in_as_admin
+    an_agent_exists_with_username
+  end
+
+  scenario 'User clicks the New Sponsor button' do
+    when_i_register_new_sponsor
+
+    then_the_sponsor_should_be_saved
+    and_i_should_be_on :new_sponsor_page
```

Comment:
How about something like this:
```
#new
when_i_fill_in_new_sponsor_form_for "Paul"    #when_i_register_new_sponsor
click_button 'Create and Add Another' 
and_i_should_be_on :new_sponsor_page
then_the_sponsor_should_be_saved_for "Paul"   (not sure if this check should be part of integration tests)

when_i_fill_in_new_sponsor_form_for "Tom"
click_button 'Create'
then_the_sponsor_should_be_saved_for "Tom"   (not sure if this check should be part of integration tests)
and_i_should_be_on :sponsor_page, "Tom"
and_i_should_see_details_for "Tom"

#index
visit :sponsors_path
and_i_should_see "Paul"
and_i_should_see "Tom"
```

something like this will exercise `create and add another`, `create`, show view and index view

Also we have to add the `#edit` here too: Click Edit for Paul, change some fields (all would be best) and check show view for changed fields. It might be better to put `#edit` in a different scenario to keep them smaller.